### PR TITLE
Refactor Menu v2

### PR DESCRIPTION
**Pull request by weisJ**
I will PR in batches the changes from #647. Some of the changes will need some adjustment. Lets start with extracting the preview logic into its own controller class.

### CODE REVIEW

File: Maccy/Menu/Menu.swift [hunk 1]
```diff
@@ -635,12 +594,6 @@ class Menu: NSMenu, NSMenuDelegate {
     removeItem(item)
   }
 
-  private func offloadCurrentPreview() {
-    previewThrottle.cancel()
-    previewPopover?.close()
-    previewPopover = nil
-  }
-
   private func boundsOfMenuItem(_ item: NSMenuItem, _ windowContentView: NSView) -> NSRect? {
```
Comment (p0deje) [inline Maccy/Menu/Menu.swift:597]: I think you forgot to remove this one as it's now in controller.

**Comment by weisJ** [comment]
Good catch. Looks like I forgot to stage this change. Will amend it later today.

**Review by p0deje — APPROVED** [review]
LGTM, I am not sure why CI is failing but given it's green on master, can you please double-check that you haven't accidentally broken anything with this change?

**Comment by weisJ** [comment]
Should be good now

**Pull request merged**